Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -835,20 +835,36 @@ public final class PackageBuilder {
         return conditions
     }
 
+    // When building tests on macOS, we are matching the minimum deployment target of the XCTest framework.
+    func xcTestMinimumDeploymentTarget() -> PlatformVersion {
+        if let minDeploymentTarget = _xcTestMinimumDeploymentTarget {
+            return minDeploymentTarget
+        }
+        _xcTestMinimumDeploymentTarget = computeXCTestMinimumDeploymentTarget()
+        return _xcTestMinimumDeploymentTarget!
+    }
+    private var _xcTestMinimumDeploymentTarget: PlatformVersion? = nil
```

Comment:
Can you cache this in a static variable as we run the package builder for each package.